ChatClient.go

```diff
@@ -59,6 +61,26 @@ func (user *ChatClient) receive() {
 		message := *user.connection.receive()
 		parts := strings.Split(message, " ")
 		methodName := strings.Title(parts[0][1:])
+
+		// Try to avoid using reflection if possible, it's very, very slow and you lose
```

Comment:
really appreciate the explanation here. using reflection was very much intentional. I was trying to go for an analog to calling a method by name on an object as is common in js, and I wanted to explore reflection in Go at the same time. this was unnecessary true, but it had a purpose :)

Reply:
Haha yeah, I figured you might be doing it on purpose which is why I added that last bit - I haven't actually touched reflection at all yet so you're ahead of me in that regard.